service/resource/cloudformation/mocks.go

```diff
@@ -0,0 +1,58 @@
+package cloudformation
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	awscloudformation "github.com/aws/aws-sdk-go/service/cloudformation"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/iam"
+)
+
+type eC2ClientMock struct {
+	sgExists bool
+	sgID     string
+}
+
+func (e *eC2ClientMock) DescribeSecurityGroups(input *ec2.DescribeSecurityGroupsInput) (*ec2.DescribeSecurityGroupsOutput, error) {
```

Comment:
Having all the mocks in a single file is nice. I also like that the interface is small and DescribeSecurityGroups can be removed later and we can use a ref in the CF template.